YunoHost/yunohost
# Fix dyndns madness

## The problem

- Dyndns update happens way too often, every two fucking minutes is fucking madness
- If any ~long yunohost operation is ongoing, [dyndns updates command will queue up every 2 fucking minutes creating a huge mess](https://forum.yunohost.org/t/there-is-always-something-running/14003/5) and slowness all over the place
- Removing a dyndns doesnt remove the dyndns keys, then you can re-add a dyndns domain
- [What the actual fuck why was all this not already fixed 5 years ago](https://randomoverload.org/wp-content/uploads/2016/08/a9ech72150388.jpg)

## Solution

- Handle the dyndns cron from the regenconf
- Set the cron job to every 10 minutes
- Make sure that it doesn't queue up a dyndns command if there's already a yunohost command running (c.f. lock)
- Remove dyndns keys when removing a domain

## PR Status

Not really tested, gotta do a full test of all 

## How to test

Add a domain, subscribe it, check the "at" command do work, check the cron job work, check domain deletion removes the key
